src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
+    this->actor.world.pos.y = this->actor.home.pos.y - 9.0f;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    EnKusa_SetScaleSmall(this);
+    this->timer = 0;
+    this->actor.shape.rot = this->actor.home.rot;
+    this->actionFunc = EnKusa_UprootedWaitRegrow;
+}
+
+/**
+ * @brief Window of 50 frames before the grass surfaces to then regrow
+ *
+ */
+void EnKusa_UprootedWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) > 120) {
+        if ((Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.6f)) && (this->timer >= 170)) {
+            EnKusa_SetupRegrow(this);
+        }
+    }
+}
+
+void EnKusa_SetupRegrow(EnKusa* this) {
+    this->actionFunc = EnKusa_Regrow;
+    EnKusa_SetScaleSmall(this);
+    this->isCut = false;
+    this->actor.shape.rot = this->actor.home.rot;
+}
+
+void EnKusa_Regrow(EnKusa* this, GlobalContext* globalCtx) {
+    s32 isFullyGrown;
+
+    isFullyGrown = Math_StepToF(&this->actor.scale.y, 0.4f, 0.014f) & 1;
+    isFullyGrown &= Math_StepToF(&this->actor.scale.x, 0.4f, 0.011f);
+    this->actor.scale.z = this->actor.scale.x;
+    if (isFullyGrown) {
+        Actor_SetScale(&this->actor, 0.4f);
+        EnKusa_SetupMain(this);
+        this->collider.base.ocFlags1 &= ~OC1_TYPE_PLAYER;
+    }
+}
+
+void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnKusa* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->isCut != 0) {
+        this->actor.shape.yOffset = -6.25f;
+    } else {
+        this->actor.shape.yOffset = 0.0f;
+    }
+    if ((kusaGameplayFrames != globalCtx->gameplayFrames) && (globalCtx->roomCtx.currRoom.unk3 == 0)) {
+        EnKusa_Sway();
+        kusaGameplayFrames = globalCtx->gameplayFrames;
+    }
+}
+
+void EnKusa_DrawBush(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnKusa* this = THIS;
+
+    if ((this->actor.projectedPos.z <= 1200.0f) || ((this->unk198 & 1) && (this->actor.projectedPos.z < 1300.0f))) {
+        if ((globalCtx->roomCtx.currRoom.unk3 == 0) && (this->actionFunc == EnKusa_Main) &&
+            (this->actor.projectedPos.z > -150.0f) && (this->actor.projectedPos.z < 400.0f)) {
+            EnKusa_GetMtxState(&D_80936AD8[this->kusaMtxIdx]);
+        }
+        Gfx_DrawDListOpa(globalCtx, gameplay_field_keep_DL_0078A0);
+    } else if (this->actor.projectedPos.z < 1300.0f) {
+        s32 alpha;
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        alpha = (1300.0f - this->actor.projectedPos.z) * 2.55f;
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, alpha);
+        gSPDisplayList(POLY_XLU_DISP++, gameplay_field_keep_DL_007938);
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
```

Comment:
I'd space this out rather more.
```suggestion
        s32 alpha;

        OPEN_DISPS(globalCtx->state.gfxCtx);

        alpha = (1300.0f - this->actor.projectedPos.z) * 2.55f;
        func_8012C2DC(globalCtx->state.gfxCtx);
        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, alpha);
        gSPDisplayList(POLY_XLU_DISP++, gameplay_field_keep_DL_007938);

        CLOSE_DISPS(globalCtx->state.gfxCtx);
```
And document that displaylist.